### PR TITLE
Add ability to send text lines via serial

### LINKED ISSUE #1622: As a xoder I want to send text lines via serial from debugger tab to give quick board commands
### Rationale

A very basic way to talk to the board from PC is with plain-text serial commands. The [debugger protocol](https://github.com/xodio/xod/wiki/Debugger-protocol) is based on them as well. Currently, there’s no way to send anything custom from within IDE and simple line-by-line transmission would be enough for quick xoding experiments.

### User story

- I upload my program to the board
- If I upload with the debugger, I stop it
- I hit “Deploy → Connect Serial”
- A dialog box is shown asking for the port name, I select the port and press Connect
- The deployment pane pops up with the tab switched to Serial
- I type custom text to the input box and hit Enter
- The line gets cleared, the text is pushed through the serial connection with `\r\n` added to the tail automatically
- The line is echoed in the serial log with `↖` symbol
- Incoming lines are added to the log prefixed with the `↘` symbol
- I press the disconnect button and the serial session ends

### Acceptance criteria

* [ ] The basic scenario works as described above
* [ ] I can send a line with Enter or by clicking a “Send” button next to it
* [ ] The current “Debugger” tab is renamed to “Serial”
* [ ] The lines sent are mixed with the lines received: if the log is not followed, I don’t see them until start to follow the log
* [ ] Trying to upload or connect to an already connected serial port stops the existing connection first

### PR DESCRIPTION
**Pull request by evgenykochetkov**
Closes #1622

### CODE REVIEW

File: packages/xod-client-electron/src/shared/events.js [hunk 1]
```diff
@@ -34,8 +34,11 @@ export const APP_UPDATE_DOWNLOAD_REQUEST = 'EVENT_APP_UPDATE_DOWNLOAD_REQUEST';
 export const APP_UPDATE_DOWNLOAD_STARTED = 'EVENT_APP_UPDATE_DOWNLOAD_STARTED';
 
 export const START_DEBUG_SESSION = 'START_DEBUG_SESSION';
-export const DEBUG_SESSION = 'DEBUG_SESSION';
+export const SEND_TO_DEBUG_SERIAL = 'SEND_TO_DEBUG_SERIAL';
+export const MESSAGES_FROM_SERIAL_SESSION = 'MESSAGES_FROM_SERIAL_SESSION';
```
Comment (nkrkv) [inline packages/xod-client-electron/src/shared/events.js]: The proper form of AT name is `<NOUN>_<VERBINFINITIVE>` or consistent to surroundings. This seems like neither of cases.

Should it better be:

- `DEBUG_SERIAL_SEND`
- `SERIAL_SESSION_MESSAGE_RECEIVE` (or `SERIAL_SESSION_RECEIVE`)

File: packages/xod-client-electron/src/upload/components/PortSelect.jsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  selectedPort: PropTypes.object,
+  listPorts: PropTypes.func,
+  onPortChanged: PropTypes.func,
+};
+
+PortSelect.defaultProps = {
+  isDeploymentInProgress: false,
+};
+
+export default PortSelect;
```
Comment (nkrkv) [inline packages/xod-client-electron/src/upload/components/PortSelect.jsx]: :+1: For extracting to a module

**Comment by nkrkv** [comment]
Also, I think a dark theme for the serial input will work nicer. A rough sketch:

image

The background color is the same used in the C++ editor (#3a3a40). The border is ~#555555.

Also, note please, that the input is not attached to the very bottom. I see a 2-3px gap there.

**Comment by brusherru** [comment]
Also, I think it will be nice to make possible to select baud rate in the popup...

**Comment by evgenykochetkov** [comment]
@nkrkv @brusherru Fixed! Check again, please

**Pull request merged**